src/coreclr/src/inc/clrconfigvalues.h

```diff
@@ -710,10 +710,10 @@ RETAIL_CONFIG_DWORD_INFO(INTERNAL_EventPipeCircularMB, W("EventPipeCircularMB"),
 RETAIL_CONFIG_DWORD_INFO(INTERNAL_EventPipeProcNumbers, W("EventPipeProcNumbers"), 0, "Enable/disable capturing processor numbers in EventPipe event headers")
 
 //
-// Diagnostics Server
+// Diagnostics Ports
 //
-RETAIL_CONFIG_STRING_INFO_EX(EXTERNAL_DOTNET_DiagnosticsMonitorAddress, W("DOTNET_DiagnosticsMonitorAddress"), "NamedPipe path without '\\\\.\\pipe\\' on Windows; Full path of Unix Domain Socket on Linux/Unix.  Used for Diagnostics Monitoring Agents.", CLRConfig::DontPrependCOMPlus_);
-RETAIL_CONFIG_DWORD_INFO_EX(EXTERNAL_DOTNET_DiagnosticsMonitorPauseOnStart, W("DOTNET_DiagnosticsMonitorPauseOnStart"), 1, "If DOTNET_DiagnosticsMonitorAddress is set, this will cause the runtime to pause during startup.  Resume using the Diagnostics IPC ResumeStartup command.", CLRConfig::DontPrependCOMPlus_);
+RETAIL_CONFIG_DWORD_INFO_EX(EXTERNAL_DOTNET_DiagnosticPortSuspend, W("DOTNET_DiagnosticPortSuspend"), 0, "This will cause the runtime to pause during startup before major subsystems are started.  Resume using the Diagnostics IPC ResumeStartup command.", CLRConfig::DontPrependCOMPlus_);
+RETAIL_CONFIG_STRING_INFO_EX(EXTERNAL_DOTNET_DiagnosticPorts, W("DOTNET_DiagnosticPorts"), "A semicolon delimited list of additional Diagnostic Ports, where a Diagnostic Port is a NamedPipe path without '\\\\.\\pipe\\' on Windows or the full path of Unix Domain Socket on Linux/Unix followed by optional tags, e.g., '<path>,listen,nosuspend;<path>,connect'", CLRConfig::DontPrependCOMPlus_);
```

Comment:
One problem that I've been thinking about is if the path portion of the port description is a full path, then there are likely to be collisions if multiple processes see the same port description (because this is provided by an environment variable). At best, there is a race to see which process can establish its endpoint first and all of the others will silently fail. I'm wondering if the path portion needs to be the "directory" of where to establish the pipe or socket file and then the runtime uses its default naming convention of the file itself.

Reply:
That's certainly possible for a `listen` port, but I'm not sure we would want users doing that.  That scenario was the impetus for reversing the pipe in the first place.  I Think it would be fair to simply document that if you specify a `listen` port, it should be specified for only a single process.  If you are in a multi-process scenario, I would expect users to only use `connect` ports.

Reply:
You don't necessarily have control of what the process spawns. Third party libraries could spawn their own .NET process and very likely don't know to disable diagnostics on them. Even "dotnet run" spawns multiple processes.

Reply:
I'm comfortable just not allowing `listen` ports to be specified other than the default one.

Reply:
I talked with @tommcdon offline, and we decided it is safer to just disable the ability to specify additional `listen` ports for now.  If later we decide to re-enable it, then we can add limitations or specifiers to the name to solve this problem.

Reply:
Just for clarification, that means we'll remove the "listen" and "connect" tags since all ports are connect mode?

Reply:
I'll leave the code for consuming it but ignore the tag if it's seen.  It'll still be legal to specify `connect`, I'll just ignore `listen` and make the default `connect`.

Reply:
The issue text up top appears inconsistent now btw, in some places it has been updated to be connect-only, in others it still suggests listen is supported.

Reply:
Updated the issue text.